### DISCUSSION TITLE
Some feedback and ideas

### DISCUSSION THREAD

**Discussion by T9es** [HRNPH/AIwaifu#38] [Ideas]
So, I've played around for a few hours now with the model and the whole setup. Here are my notes:

```
- Optimize the code for remembering

I saw this already being touched upon, but this is a big one, could be really useful. 
Something like perma remember would be also good, a list o things that ALWAYS will be remembered
```
.

```
- Build a simple localhost GUI for easier manipulation, on top of the config file that you are planning to do

This is nothing of importance, just a QOL change that would be cool 
to see in the long run if you plan on dragging this project along
```
.

```
- Make the model a bit smarter

Currently the model gets confused easily. 
When asking for the 4 first digits of PI, the model just says: (3....). 
After asking the next digit, it said 7 for me, so that's random. 
Screenshot at the bottom


Would be cool to see an option to load a chat-gpt like intelligence into it, or make the msg remember code so optimized, 
that it can go through thousands of saved ones without a hustle, which I guess would be hard to do. 

There were also multiple instances where I managed to freeze the AI.
By freeze, I mean it got so confused that it started repeating it's own words over and over.
What did I ask it that it froze? Well, nothing of importance. Casual conversation really, don't have screenshots,
since this was yesterday when I was sleepwalking 

Currently, there are multiple models avalible, like Alpaca that could work for this, however, 
I have no idea how that could be implemented. 
I don't code in python, or in any advanced setting, so treat this with a grain of salt if it's hard or impossible to implement.

"the goal is to just giving everyone a foudational platform to develop their own waifu"
That's the main goal of the project, however, people without prior coding knowledge in python will be confused, 
especially considering this is AI territory, so even more confusing for newcomers
```
![image](https://user-images.githubusercontent.com/107748212/235303334-785df0ec-df3e-4896-9f38-4d470e43f861.png)
.

```
- Options to use a built in translator into english, to then make the AI respond (google translate, or DeepL like translators)

Currently as far as I can see, english is the language that MOST people use, but there are many others. 
For twitch streaming (if you are planning to go this far)
only english will be hard, so it would be nice to see this feature somewhere along the way.
```
.

```
- Make API integrations easy with youtube chat or twitch chat, or even kick chat ( kick to avoid filtering in the future, avoid bans etc.)

Since this is a long shot, the project is "inspired by neuro-sama". 
I doubt it would be hard to implement this, as multiple people have already done this on github, 
and I can see someone streaming this for fun while they play with friends or something
```

There are multiple other things, like easily changing the TTS voice language to english (easy way, instead of going through code) or other stuff, but that will be possible along the way I suppose. Making the model smarter from all the ideas would be the BIGGEST change

**Comment by HRNPH** [comment]
Agree with you on many terms especially on _**Ease Of Use**_ currently it it REALLY HARD to set up and play with
but this is what we'll fix later on after every function is stable

an optimization for chatting experience will be working with later on after the TTS Corpus(which is soon finishing)
it is not impossible to implement Top-Notch opensource Language Model (OpenAssistant Alphaca LLAMA)
into the projects however it just took too much ram for basically any computer that isn't a GPU cluster
we'll be better of using ChatGPT API if we need High-performance GPU for the model to work

a localhost GUI for easier manipulation WILL BE implemented but I don't know when yet
since I plan to add many many functionality later such as Gaming chatting and stuff so I don't yet design how the control panel should be

Options to use a built-in translator in English Yeah I think you got a good point, Speaking in English Right?
this can actually be done and I think it'll be our next Issue to make it multilingual support (for speaking at least)
Fun-Fact (I actually translate English -> Japanese because I want Japanese sound lol)

there's a Fork Working on Twitch/Youtube Integration chat I don't know when will the Pull Requests came thou lol
so I think I'll implement this myself it is not that hard but this should be done after the control panel was done

**Anyway, thanks for the feedback it did helps a lot!**
if you have more suggestions feel free to comment more! I always read (even sometime I don't answer)
